assets/js/theme/common/product-details.js

```diff
@@ -35,8 +35,24 @@ export default class Product {
         this.imageGallery.init();
         this.listenQuantityChange();
         this.initRadioAttributes();
-        this.updateProductAttributes(productAttributesData);
-        this.updateView(productAttributesData);
+
+        const $form = $('form[data-cart-item-add]', $scope);
+        const $hasOptions = $('[data-product-option-change]', $form).html().trim().length;
+        const $isQuickView = $scope.hasClass('quickView');
+
+        // Update product attributes. If we're in quick view and the product has options, then also update the initial view in case items are oos
+        if ($isQuickView && $hasOptions && window.FormData !== undefined) {
```

Comment:
I'll change to checking if `window.BCData.product_attributes` is undefined, but will still need to check if there are options because the `product-attributes` endpoint only returns data if the product has options.